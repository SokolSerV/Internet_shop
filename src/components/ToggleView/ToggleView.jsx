import { useContext } from "react";

import ViewContext from "../../context/ViewContext";

import { ReactComponent as ToggleOffIcon } from "./icons/toggle-off.svg";
import { ReactComponent as ToggleOnIcon } from "./icons/toggle-on.svg";

export default function ToggleView() {
    const viewData = useContext(ViewContext);
    const { view, toggleView } = viewData;

    return (
        <div
            style={{
                display: "flex",
                justifyContent: "end",
                alignItems: "center",
                margin: "5px 50px",
            }}
        >
            <p style={{ margin: "0 10px 0 0 ", fontWeight: "700" }}>
                Показати у вигляді {view === "card" ? "таблиці" : "карток"}
            </p>
            <i
                onClick={toggleView}
                style={{ display: "flex", cursor: "pointer" }}
            >
                {view === "card" ? <ToggleOffIcon /> : <ToggleOnIcon />}
            </i>
        </div>
    );
}
