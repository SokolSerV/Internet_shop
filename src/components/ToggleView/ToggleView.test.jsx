import { render } from "@testing-library/react";

import ToggleView from "./ToggleView";

describe("ToggleView component Smoke test (Snapshot test)", () => {
    test("should ToggleView render", () => {
        const { asFragment } = render(<ToggleView />);

        expect(asFragment()).toMatchSnapshot();
    });
});
