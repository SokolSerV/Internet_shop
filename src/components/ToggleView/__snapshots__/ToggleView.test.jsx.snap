// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleView component Smoke test (Snapshot test) should ToggleView render 1`] = `
<DocumentFragment>
  <div
    style="display: flex; justify-content: end; align-items: center; margin: 5px 50px;"
  >
    <p
      style="margin: 0px 10px 0px 0px; font-weight: 700;"
    >
      Показати у вигляді карток
    </p>
    <i
      style="display: flex; cursor: pointer;"
    >
      <svg>
        toggle-on.svg
      </svg>
    </i>
  </div>
</DocumentFragment>
`;
